### DISCUSSION TITLE
Is there any reason why middleware is limited to specific web apis when not being used on edge functions?

### DISCUSSION THREAD

**Discussion by Markos-Th09** [vercel/next.js#34085] [Help]
Is there really any reason why middleware is limited to specific web apis when not being used on edge functions? In non serverless environments it is a bit stupid because a full server capable of running all the node apis can't use them in middleware.

**Comment by thescientist13** [comment]
Same!  

Was searching through the discussions here and saw this which looks it could be the solution to unlocking this and allowing us to opt-out of the edge run time.  🤞 
https://github.com/vercel/next.js/discussions/34179

**Reply by Markos-Th09** [reply]
> Same!
> 
> Was searching through the discussions here and saw this which looks it could be the solution to unlocking this and allowing us to opt-out of the edge run time. 🤞 #34179

If they add it for middleware it would be amazing
